source/iterator/snapshot.go

```diff
@@ -216,9 +220,33 @@ func (iter *Snapshot) Close() error {
 	return nil
 }
 
+// checkIfTableExists checks if table exist.
+func (iter *Snapshot) checkIfTableExists(ctx context.Context, tx *sql.Tx, table string) (bool, error) {
+	rows, err := tx.QueryContext(ctx, fmt.Sprintf(queryIfTableExists, table))
+	if err != nil {
+		return false, fmt.Errorf("request with check if the table already exists: %w", err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var name string
+		err = rows.Scan(&name)
+		if err != nil {
+			return false, fmt.Errorf("scan tables to check if the table already exists: %w", err)
+		}
+
+		if name == table {
```

Comment:
Is there a special reason this is needed, because we're already doing it in the query?

Reply:
Correct.
I've removed this unnecessary check.
👍